### PR TITLE
Create Johnson's-algo.md

### PR DESCRIPTION
**Pull request by pranaydeep1**
This PR adds a README file for Johnson's Algorithm, explaining the algorithm, its time and space complexity, and common applications. This documentation helps users understand the implementation and use cases of Johnson's Algorithm.

### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue number)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by pranaydeep1** [comment]
@ajay-dhangar  can you review and approve my pull request as tomorrow is last date for gssoc'24 extd

File: docs/graphs/Johnson's-algo.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+### Example:
+#### Input:
+- Graph vertices: 4
+- Edges:  
+  - (0 -> 1, weight=3)  
+  - (0 -> 2, weight=8)  
+  - (1 -> 3, weight=-2)  
+  - (2 -> 1, weight=4)  
+
```
Comment (ajay-dhangar) [inline docs/graphs/Johnson's-algo.md]: write within ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by pranaydeep1** [comment]
@ajay-dhangar  now i have updated, sorry for that . Kindly review

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**